src/components/form/form.component.js

```diff
@@ -0,0 +1,163 @@
+  StyledSaveContainer
+} from './form.style';
+
+const SCROLL_THROTTLE = 100;
+
+const Form = ({
+  children,
+  saveButton,
+  leftSideButtons,
+  rightSideButtons,
+  errorCount,
+  warningCount,
+  onSubmit,
+  buttonAlignment,
+  stickyFooter,
+  showSummary,
+  ...rest
+}) => {
+  const [isFooterSticky, setIsFooterSticky] = useState(false);
+
+  const formRef = useRef();
+
+  const formFooterRef = useRef();
+
+  const stickyListenersAddedRef = useRef(false);
+
+  const checkStickyFooter = useCallback(throttle(() => {
+    const formHeight = (formRef.current.offsetTop + formRef.current.offsetHeight) - window.pageYOffset;
+    const footerHeight = formFooterRef.current.offsetHeight;
+
+    if (formHeight - (footerHeight / 2) > window.innerHeight) {
+      setIsFooterSticky(true);
+    } else if (formHeight < window.innerHeight) {
+      setIsFooterSticky(false);
+    }
+  }, SCROLL_THROTTLE), []);
+
+  const addStickyFooterListeners = useCallback(() => {
+    ElementResize.addListener(formRef.current, checkStickyFooter);
+    window.addEventListener('resize', checkStickyFooter, true);
+    window.addEventListener('scroll', checkStickyFooter, true);
+  }, [checkStickyFooter]);
+
+  const removeStickyFooterListeners = useCallback(() => {
+    ElementResize.removeListener(formRef.current, checkStickyFooter);
+    window.removeEventListener('resize', checkStickyFooter, true);
+    window.removeEventListener('scroll', checkStickyFooter, true);
+  }, [checkStickyFooter]);
+
+  useEffect(() => {
+    if (stickyFooter && !stickyListenersAddedRef.current) {
+      addStickyFooterListeners();
+      checkStickyFooter();
+      stickyListenersAddedRef.current = true;
+    }
+    return () => removeStickyFooterListeners();
+  }, [addStickyFooterListeners, checkStickyFooter, removeStickyFooterListeners, stickyFooter]);
+
+  const saveButtonWithSummary = () => {
+    const save = (
+      <StyledSaveContainer>
+        { saveButton }
+      </StyledSaveContainer>
+    );
+    if (showSummary) {
+      return <FormSummary errors={ errorCount } warnings={ warningCount }>{save}</FormSummary>;
+    }
+    return save;
+  };
+
+  const formButtons = () => {
+    const buttons = (
+      <>
+        {leftSideButtons && (
+          <StyledLeftButtons buttonAlignment={ buttonAlignment }>
+            { leftSideButtons }
+          </StyledLeftButtons>
+        )}
+        {saveButtonWithSummary()}
+        {rightSideButtons && (
+          <StyledRightButtons buttonAlignment={ buttonAlignment }>
+            { rightSideButtons }
+          </StyledRightButtons>
+        )}
+      </>
+    );
+    return buttons;
+  };
+
+  return (
+    <StyledForm
+      ref={ formRef }
+      stickyFooter={ stickyFooter && isFooterSticky }
+      onSubmit={ onSubmit }
+      data-component='form'
+      { ...rest }
+    >
+      { children }
+      <StyledFormFooter
+        data-element='form-footer'
+        className={ isFooterSticky ? 'isSticky' : '' }
```

Comment:
can we do this with a prop or do we need `.isSticky` for Dialog?

Reply:
I need to override `StyledFormFooter` style when `Form` is rendered in `Dialog` component but only when the form footer is in sticky state. Adding `isSticky` class to `StyledFormFooter` is the easiest way to achieve it since I don't have access to its props in Dialog styles.